examples/calling-apis/langchain-examples/src/app/app.py

```diff
@@ -1,104 +1,187 @@
+from fastapi import FastAPI, Request, Response, HTTPException, Depends
+from fastapi.responses import JSONResponse, RedirectResponse
+from fastapi.templating import Jinja2Templates
+from pathlib import Path
+from starlette.middleware.sessions import SessionMiddleware
+from auth_types import StartInteractiveLoginOptions
+
+from auth0_fastapi.server.routes import router, register_auth_routes
+
 from langgraph_sdk import get_client
+from langgraph_sdk.schema import Command
 from langchain_core.messages import HumanMessage
 from auth0_ai_langchain.utils.interrupt import get_auth0_interrupts
+from src.auth0.auth import config, auth_client
+
 
 load_dotenv()
 
-app = Flask(__name__)
-app.secret_key = os.getenv("APP_SECRET_KEY", "YOUR_SECRET_KEY")
-
-oauth = OAuth(app)
-oauth.register(
-    "auth0",
-    client_id=os.getenv("AUTH0_CLIENT_ID"),
-    client_secret=os.getenv("AUTH0_CLIENT_SECRET"),
-    client_kwargs={
-        "scope": "openid profile offline_access",
-    },
-    server_metadata_url=f'https://{os.getenv("AUTH0_DOMAIN")}/.well-known/openid-configuration'
-)
-
-@app.route("/")
-def home():
-    if "user" not in session:
-        return redirect("/login")
-
-    session.pop("thread_id", None)
-    session.pop("messages", None)
-    return render_template("index.html", user=session.get('user'))
-
-@app.route("/chat", methods=["POST"])
-async def chat():
-    if "user" not in session:
-        return jsonify({"error": "unauthorized"}), 401
-    
-    if "messages" not in session:
-        session["messages"] = []
-    
-    message = request.json.get("message")
-    session["messages"].append(message)
-
-    client = get_client(url=os.getenv("LANGGRAPH_API_URL", "http://localhost:54367"))
-
-    async def run_and_return():
-        if "thread_id" not in session:
-            session["thread_id"] = (await client.threads.create())["thread_id"]
-
-        return await client.runs.wait(
-            session["thread_id"],
+app = FastAPI(
+    title="Auth0 AI + LangChain - Chatbot Example: Calling API's on user's behalf")
+
+app.add_middleware(SessionMiddleware, secret_key=os.getenv(
+    "APP_SECRET_KEY", "SOME_RANDOM_SECRET_KEY"))
+
+# Attach to the FastAPI app state so internal routes can access it
+app.state.config = config
+app.state.auth_client = auth_client
+
+# Conditionally register routes
+register_auth_routes(router, config)
+
+# Include the SDK’s default routes
+app.include_router(router)
+
+# Set up templates directory with custom Jinja2 environment
+templates = Jinja2Templates(directory=Path(__file__).parent / "templates")
+
+# Initialize LangGraph client
+langgraph_client = get_client(url=os.getenv(
+    "LANGGRAPH_API_URL", "http://localhost:54367"))
+
+
+@app.get("/")
+async def home(request: Request, response: Response):
+    # Check if user is authenticated
+    try:
+        await auth_client.require_session(request, response)
+    except Exception as e:
+        return RedirectResponse(url="/auth/login")
```

Comment:
`/auth/login` or `/auth/signin` ?

Reply:
> When the auth0_server_python framework is updated, I shall just remove the endpoint and call /auth/login

ok, consider adding a comment explaining that and maybe rename the endpoint to differentiate it from `/auth/login`